frontend/packages/ceph-storage-plugin/src/components/modals/volume-snapshot-modal/volume-snapshot-modal.tsx

```diff
@@ -29,17 +36,130 @@ export type VolumeSnapshotModalProps = {
 export const VolumeSnapshotModal = withHandlePromise((props: VolumeSnapshotModalProps) => {
   const { close, cancel, pvcData, errorMessage, inProgress, handlePromise } = props;
   const resource = pvcData.data as K8sResourceKind;
+  const [snapshotCount, setSnapshotCount] = React.useState(3);
   const [snapshotName, setSnapshotName] = React.useState(
     `${getName(resource) || 'pvc'}-snapshot-1`,
   );
+  const [scheduleLabel, setScheduleLabel] = React.useState('');
   const snapshotTypes = {
-    single: 'Single Snapshot',
+    SingleSnapshot: 'SingleSnapshot',
+    CronJob: 'CronJob',
   };
+  const selectedKey = snapshotTypes.SingleSnapshot;
+  const [scheduleType, setScheduleType] = React.useState('');
 
+  const setSchedulerHandler = (value) => {
+    setScheduleType(snapshotTypes[value]);
+  };
+
+  const getCronElements = () => {
+    const numberOfSnapshot = {
+      '3': '3',
+      '5': '5',
+      '7': '7',
+    };
+    const initialCountKey = numberOfSnapshot['3'];
+    const setSchedulerCountHandler = (value) => {
+      setSnapshotCount(parseInt(value, 10));
+    };
+    return (
+      <div>
+        <FormGroup className="ceph-volume-snapshot-modal--input" fieldId="snapshot-modal__schedule">
```

Comment:
```suggestion
        <FormGroup className="ceph-volume-snapshot-modal--input" fieldId="snapshot-modal__schedule" label={'Label}>
```
label prop takes in ReactNode type so you can pass anchors too if required.